### ISSUE TITLE
CsvConfiguration second constructor automatically sets IsNewLineSet to true

### ISSUE THREAD

**Issue by jbarrau** [JoshClose/CsvHelper#1685]
**Describe the bug**

Hello,

When upgrading from 19.0 to 22.1 I faced errors in my unit tests. After investigation it turned out it was because I was using the constructor with many parameters which changed the behaviour related to new line handling.
 
Before I was using:
```
var configuration = new CsvConfiguration(CultureInfo.InvariantCulture);
configuration.HasHeaderRecord = false;
```
With the removal of the configuration field setters I replaced this code with:
```
var configuration = new CsvConfiguration(CultureInfo.InvariantCulture, hasHeaderRecord: false);
```
However this had the unexpected side effect to set `IsNewLineSet` to true, which is not what we would expect from this constructor... It seems to me that `IsNewLineSet` should be kept to false by default, unless the caller explicitly specified a new line sequence.
The problem seems to come from here:
https://github.com/JoshClose/CsvHelper/blob/4db081ab48a08d3424fb663fddf934769efbbd91/src/CsvHelper/Configuration/CsvConfiguration.cs#L282
where we would rather expect `newLine = newLine ?? NewLine`.

**To Reproduce**

```
var configuration = new CsvConfiguration(
                    cultureInfo: CsvUtils.CsvCulture,
                    hasHeaderRecord: false);
```
=> results in `IsNewLineSet` true (which is unexpected: the caller hasn't specified any new line sequence)
```
var configuration = new CsvConfiguration(CsvUtils.CsvCulture)
                {
                    HasHeaderRecord = false
                }
```
=> results in `IsNewLineSet` false (expected behaviour)

**Comment by JoshClose** [comment]
Fixed in 22.1.1 on NuGet. https://www.nuget.org/packages/CsvHelper/22.1.1

**Comment by jbarrau** [comment]
Thanks!
